enhancements/user-defined-default-ingress-controller.md

```diff
@@ -0,0 +1,262 @@
+
+1. Install a cluster which unconditionally exposes ingress with a public load
+   balancer.
+2. Replace the default IngressController with an internal variant.
+
+This replacement procedure violates the constraints of the goal (as a public
+load balancer exists for some time). The procedure is also disruptive:
+
+1. During the replacement operation, default ingress is completely uninstalled
+   and then reinstalled, resulting in complete (but temporary) ingress
+   disruption.
+    * The old cloud load balancer is destroyed and a new load balancer is
+      provisioned.
+    * DNS records for the old cloud load balancer are destroyed and then
+      re-created for the new load balancer.
+
+If this proposal is implemented, during installation the user can provide the
+following IngressController manifest which satisfies the constraint as well as
+eliminating disruption and unnecessary (and possibly expensive) cloud resource
+provisioning.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: default
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+```
+
+#### Pre-configuring shards
+
+>As an OpenShift administrator, I want to the default IngressController to only
+>admits OpenShift's default Routes
+
+To accomplish this goal today, users must:
+
+1. Install a cluster which by default produces an IngressController which will
+   admit any Route resource in any namespace.
+2. Update the default IngressController to specify Route selector criteria which
+   exclude user defined Routes (where user defined Route is a Route labeled to
+   indicate the Route is user-defined).
+
+The update procedure has at least two notable drawbacks:
+
+1. Changing the IngressController will result in a rollout (which today can be
+   disruptive).
+2. The IngressController may already have admitted some Routes which would have
+   been excluded had the IngressController been created with the desired
+   selection criteria.
+
+If this proposal is implemented, during installation the user can provide an
+IngressController manifest which has the desired Route selection criteria,
+eliminating both drawbacks.
+
+### Implementation Details/Notes/Constraints
+
+The IngressController resource is a cluster-scoped Custom Resource Definition.
+For the cluster to support creation of the default IngressController resource,
+the following must be true:
+
+1. The IngressController Custom Resource Definition must be persisted and
+   admitted into the cluster.
+2. The `openshift-ingress-operator` namespace must exist to contain the
+   `default` IngressController resource.
+
+The Custom Resource Definition and Namespace manifests are part of the Ingress
+Operator image, and are normally created and managed by the [Cluster Version Operator](https://github.com/openshift/cluster-version-operator). Creation order of these resources
```

Comment:
During install all resources all created without ordering constraints between operators.

Reply:
> During install all resources all created without ordering constraints between operators.

Pretty sure that rendered manifests are created in alphabetical order.

Reply:
By rendering the default ingress operator, the race is removed.  We already have CRDs that are both rendered and managed by the CVO (see cluster-config-operator)